openupgrade_scripts/scripts/stock/18.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2025 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_field_renames = [
+    ("stock.move", "stock_move", "location_dest_id", "location_final_id"),
```

Comment:
@MiquelRForgeFlow did you test this with a multi step route? Now working on the mrp migration, I very much doubt this is correct.

I think we need to keep location_dest_id what it is, and reconstruct location_final_id from the procurement that cause the move

Reply:
Oh, then it lacks some script to update the `location_final_id` at some point in the migration. I suppose it has to check the locations of move_dest_ids, right? Do you have any concrete idea?

Reply:
I'm now working on a function that takes all stock moves from the procurement group, and determines the "greatest" location_dest_id in the sense that there's no other stock move that goes out of this location within the group. will probably need some fine tuning re scrap locations and the like

Reply:
this is more tricky then it looked like in the beginning, will revisit next week

Reply:
```sql
UPDATE stock_move sm
SET location_final_id = COALESCE(sr.location_dest_id, sm.location_final_id)
FROM stock_rule sr
WHERE sm.rule_id = sr.id AND sm.location_dest_from_rule
```

What do you think of this?

Reply:
See https://github.com/OCA/OpenUpgrade/pull/5088.